### PR TITLE
Add ScalaTest 3.1.0/ScalaCheck 1.14 support for play25-json-tests

### PR DESCRIPTION
**Pull request by alexkamil**
add `play25-json-tests-sc14` artifact compatible with [ScalaTest 3.1.0](http://www.scalatest.org/release_notes/3.1.0) and [ScalaCheck 1.14](https://github.com/typelevel/scalacheck/blob/1.14.0/RELEASE.markdown)

The reason is that ScalaTest 3.1.0 and ScalaCheck 1.14 are not binary compatible with earlier versions of ScalaTest/ScalaCheck, and currently play-json-ops only supports these versions with Play 2.7 via `play27-json-tests-sc14` artifact (but not with Play 2.5)

what was done:

- added new artifact `play25-json-tests-sc14` in `playJsonTests` in build.sbt

- created new directory `play25-json-tests-sc14/src` by copying `play-json-tests-common/src` and updating import statements to be compatible with: 
1) ScalaTest 3.1.0 as per http://www.scalatest.org/release_notes/3.1.0
e.g. `org.scalatest.FlatSpecLike` -> `org.scalatest.flatspec.AnyFlatSpecLike`, etc.
2) ScalaCheck 1.14/[ScalaTestPlusScalaCheck 1.14](https://github.com/scalatest/scalatestplus-scalacheck/releases/tag/release-3.1.1.0-for-scalacheck-1.14)
e.g. `org.scalatest.prop.GeneratorDrivenPropertyChecks` -> `org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks`

to see the list of changes run 
` diff -bur play-json-tests-common  play25-json-tests-sc14`

- updated Readme

### CODE REVIEW

**Review by jeffmay — APPROVED** [review]

**Review by ripplekhera — APPROVED** [review]

File: play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/DurationGenerators.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  implicit val arbTimeUnit: Arbitrary[TimeUnit] = Arbitrary {
+    Gen.oneOf(
+      DAYS,
+      HOURS,
+      MICROSECONDS,
+      MILLISECONDS,
+      MINUTES,
+      NANOSECONDS,
+      SECONDS
```
Comment (usufruct99) [inline play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/DurationGenerators.scala:19]: this ordering hurts me. :)

File: play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/JsValueGenerators.scala [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  }
+
+  implicit def arbJsBoolean: Arbitrary[JsBoolean] = Arbitrary {
+    Gen.oneOf(true, false) map JsBoolean
+  }
+
+  /**
+   * The Jaxson parser has trouble with very large exponents of BigDecimals.
+   *
+   * I couldn't quite pin the problem down to precision, scale, or both, so
```
Comment (usufruct99) [inline play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/JsValueGenerators.scala:58]: please complete this comment when you get a chance.

**Review by usufruct99 — APPROVED** [review]

File: play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Depth.scala [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package play.api.libs.json.scalacheck
+
+import scala.language.implicitConversions
+
+class Depth private[Depth] (val depth: Int) extends AnyVal with Counted with Proxy {
+  override def self = depth
+  override protected def throwOnNegative(): Nothing = throw new IllegalArgumentException("Depth cannot be negative")
+  @inline override def count: Int = depth
+  def -(that: Depth) = Depth(this.depth - that.depth)
+  def +(that: Depth) = new Depth(this.depth + that.depth)  // no need to validate
```
Comment (kapkaev) [inline play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Depth.scala:10]:
```suggestion
  def +(that: Depth) = Depth(this.depth + that.depth)  // no need to validate
```

File: play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Depth.scala [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  implicit def fromInt(int: Int): Depth = Depth(int)
+
+  implicit def toInt(depth: Depth): Int = depth.depth
+
+  override def apply(depth: Int): Depth = {
+    val d = new Depth(depth)
+    d.validate()
+    d
+  }
+}
```
Comment (kapkaev) [inline play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Depth.scala]: EOF

File: play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Width.scala [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  implicit def fromInt(int: Int): Width = Width(int)
+
+  implicit def toInt(width: Width): Int = width.width
+
+  override def apply(width: Int): Width = {
+    val w = new Width(width)
+    w.validate()
+    w
+  }
+}
```
Comment (kapkaev) [inline play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Width.scala]: EOF

**Review by kapkaev — APPROVED** [review]
Is there a way to avoid copy/paste? and reuse source directory?

**Comment by alexkamil** [comment]
@kapkaev  we can't reuse [play-json-tests-common](https://github.com/rallyhealth/play-json-ops/tree/master/play-json-tests-common/src)  directory because of scalatest/scalacheck binary incompatibility, e.g. we need to change `org.scalatest.prop.GeneratorDrivenPropertyChecks` -> `org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks` but it won't work with scalacheck 1.12 only with 1.14. So there is really no choice without some major restructuring of this library, which should probably be done, but in a separate PR, this one is intended as a hotfix.

The scalatest/scalacheck upgrade was already done for play 2.7 by creating a separate [play27-json-tests-sc14](https://github.com/rallyhealth/play-json-ops/tree/master/play27-json-tests-sc14/src) directory, but we can't use it for play 2.5 as it only works with play 2.7

File: play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Width.scala (hunk 5 above)
Comment (alexkamil) [reply play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Width.scala]: fixed

File: play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Depth.scala (hunk 4 above)
Comment (alexkamil) [reply play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Depth.scala]: fixed

File: play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Depth.scala (hunk 3 above)
Comment (alexkamil) [reply play25-json-tests-sc14/src/main/scala/play/api/libs/json/scalacheck/Depth.scala:10]: i don't want to change any code in this PR except scalatest/scalacheck imports, even a minor change like this, just to be consistent with the rest of the library, so in the future if someone does refactoring via global replace for example doesn't leave out anything, as we have some overlapping pieces

**Pull request merged**